content/en/blog/_posts/2024-09-25-swap-ga-graduation/2024-09-25-swap-ga-graduation.md

```diff
@@ -0,0 +1,445 @@
+
+### Memory-backed volumes
+
+On Linux nodes, memory-backed volumes (such as [`secret`](/docs/concepts/configuration/secret/)
+volume mounts, or [`emptyDir`](/docs/concepts/storage/volumes/#emptydir) with `medium: Memory`)
+are implemented with a `tmpfs` filesystem.
+The contents of such volumes should remain in memory at all times, hence should
+not be swapped to disk.
+To ensure the contents of such volumes remain in memory, the `noswap` tmpfs option
+is being used.
+
+The Linux kernel officially supports the `noswap` option from version 6.3 (more info
+can be found in [Linux Kernel Version Requirements](/docs/reference/node/kernel-version-requirements/#requirements-other)).
+However, the different distributions often choose to backport this mount option to older
+Linux versions as well.
+
+In order to verify whether the node supports the `noswap` option, the kubelet will do the following:
+* If the kernel's version is above 6.3 then the `noswap` option will be assumed to be supported.
+* Otherwise, kubelet would try to mount a dummy tmpfs with the `noswap` option at startup.
+  If kubelet fails with an error indicating of an unknown option, `noswap` will be assumed
+  to not be supported, hence will not be used.
+  A kubelet log entry will be emitted to warn the user about memory-backed volumes might swap to disk.
+  If kubelet succeeds, the dummy tmpfs will be deleted and the `noswap` option will be used.
+  * If the `noswap` option is not supported, kubelet will emit a warning log entry,
+    then continue its execution.
+
+It is deeply encouraged to encrypt the swap space.
+See the [section above](#setting-up-encrypted-swap) with an example for setting unencrypted swap.
+However, handling encrypted swap is not within the scope of kubelet;
+rather, it is a general OS configuration concern and should be addressed at that level.
+It is the administrator's responsibility to provision encrypted swap to mitigate this risk.
+
+### Evictions
+
+Since the NodeSwap feature introduces basic swap support and leaves more advanced topics to
+follow-up features, swap-based evictions are out-of-scope for this feature.
+
+With swap being disabled, the memory eviction threshold should be set in a way that will let kubelet's
+eviction manager identify memory spikes in time so it can execute custom logic and take care of reclaiming
+memory and evicting pods before the kernel's OOM killer is invoked.
+
+However, with swap enabled, the situation is fundamentally different.
+Since swapping is a heavy operation in terms of performance, the kernel will try avoiding
+it as long as it can, and will usually start swapping only when it lacks physical memory.
+
+Therefore, when swap is enabled, it is acceptable to use all the available RAM which
+will result in the node using some of the swap memory.
+
+#### Current eviction limitations
+
+Because evictions are out-of-scope for this feature, the eviction manager would remain swap-unaware.
```

Comment:
Is this true:
```markdown
For both Linux and Windows nodes, the eviction manager is not aware of how much of the swap storage is used.
```
?

Reply:
Yes it is. I'll update this.
See my comment regarding Windows [here](https://github.com/kubernetes/website/pull/48099#issuecomment-2381288665).